docs/src/rule_author/superpowers/mutation_support.md

```diff
@@ -0,0 +1,82 @@
+# Mutation Support
+
+ChainRulesCore.jl offers experimental support for mutation, targeting use in forward mode AD.
+(Mutation support in reverse mode AD is more complicated and will likely require more changes to the interface)
+
+!!! warning "Experimental"
+    This page documents an experimental feature.
+    Expect breaking changes in minor versions while this remains.
+    It is not suitable for general use unless you are prepared to modify how you are using it each minor release.
+    It is thus suggested that if you are using it to use _tilde_ bounds on supported minor versions.
+
+
+## `MutableTangent`
+The [`MutableTangent`](@ref) type is designed to be a partner to the [`Tangent`](@ref) type, with specific support for being mutated in place.
+It is required to be a structural tangent, having one tangent for each field of the primal object.
+
+Technically, not all `mutable struct`s need to use `MutableTangent` to represent their tangents.
+Just like not all `struct`s need to use `Tangent`s.
+Common examples away from this are natural tangent types like for arrays.
+However, if one is setting up to use a custom tangent type for this it is sufficiently off the beaten path that we can not provide much guidance.
+
+## `zero_tangent`
+
+The [`zero_tangent`](@ref) function functions to give you a zero (i.e. additive identity) for any primal value.
+The [`ZeroTangent`](@ref) type also does this.
+The difference is that [`zero_tangent`](@ref) is in general full structural tangent mirroring the structure of the primal.
+To be technical the promise of [`zero_tangent`](@ref) is that it will be a value that supports mutation.
+However, in practice[^1] this is achieved through in a structural tangent
+For mutation support this is important, since it means that there is mutable memory available in the tangent to be mutated when the primal changes.
+To support this you thus need to make sure your zeros are created in various places with [`zero_tangent`](@ref) rather than []`ZeroTangent`](@ref).
+
+
+
+It is also useful for reasons of type stability, since it forces a consistent type (generally a structural tangent) for any given primal type.
+For this reason AD system implementors might chose to use this to create the tangent for all literal values they encounter, mutable or not,
+and to process the output of `frule`s to convert [`ZeroTangent`](@ref) into corresponding [`zero_tangent`](@ref)s.
+
+## Writing a frule for a mutating function
```

Comment:
Should something also be said about aliasing here? Specifically that if `a` aliases `b`, then `da` must alias `db`?

To see this, consider calling your `frule` for `foo!` below with
```julia
a = Ref(1.0)
frule((MutableTangent(x=1.0), MutableTangent(x=1.0)), foo, a, a)
```
vs
```julia
a = Ref(1.0)
da = MutableTangent(x=1.0)
frule((da, da), foo, a, a)
```
(I've not checked that the above actually runs, but you get the idea)

Reply:
Good idea

Reply:
I think my example does get it correct for that case.
Right?

Reply:
Yeah, your example is perfectly fine -- `a !== b` and `da !== db`, so we're good.

I realised that my initial statement should be stronger: it is necessary that in all situations, `a === b` <=> `da === db`. This implies that `a !== b` <=> `da !== db`.